lib/src/date-input/Calendar.tsx

```diff
@@ -153,38 +161,46 @@ const Calendar = ({
         break;
     }
   };
-
   return (
-    <CalendarContainer>
-      <DxcFlex alignItems="center" justifyContent="space-between">
+    <CalendarContainer role="grid">
+      <CalendarRow role="row">
         {weekDays.map((weekDay) => (
-          <WeekHeaderCell key={weekDay}>{weekDay}</WeekHeaderCell>
+          <CalendarCell key={weekDay} role="columnheader">
+            {weekDay}
+          </CalendarCell>
         ))}
-      </DxcFlex>
-      <DayCellsContainer onBlur={handleOnBlur}>
-        {dayCells.map((date, index) => (
-          <DayCell
-            onKeyDown={(event) => handleDayKeyboardEvent(event, date)}
-            aria-label={date.day}
-            id={`day_${date.day}_month${date.month}`}
-            key={`day_${index}`}
-            onClick={() => onDateClickHandler(date)}
-            selected={isDaySelected(date, selectedDate)}
-            actualMonth={date.month === innerDate.get("month")}
-            autoFocus={date.day === dateToFocus.get("date") && date.month === dateToFocus.get("month")}
-            aria-selected={isDaySelected(date, selectedDate)}
-            tabIndex={date.day === dateToFocus.get("date") && date.month === dateToFocus.get("month") ? 0 : -1}
-            isCurrentDay={
-              today.get("date") === date.day &&
-              today.get("month") === innerDate.get("month") &&
-              today.get("month") === date.month &&
-              today.get("year") === innerDate.get("year")
-            }
-          >
-            {date.day}
-          </DayCell>
+      </CalendarRow>
+      <WeeksContainer onBlur={handleOnBlur} role="rowgroup">
```

Comment:
We should review the naming of the containers. I think they don't describe correctly what they represent.

Reply:
Done! Take a look at it when possible.